pkg/madmin/health.go

```diff
@@ -65,36 +65,36 @@ type ServerProcInfo struct {
 
 // SysProcess - Includes process lvl information about a single process
 type SysProcess struct {
-	Pid            int32                       `json:"pid"`
-	Background     bool                        `json:"background,omitempty"`
-	CPUPercent     float64                     `json:"cpupercent,omitempty"`
-	Children       []int32                     `json:"children,omitempty"`
-	CmdLine        string                      `json:"cmd,omitempty"`
-	Connections    []nethw.ConnectionStat      `json:"connections,omitempty"`
-	CreateTime     int64                       `json:"createtime,omitempty"`
-	Cwd            string                      `json:"cwd,omitempty"`
-	Exe            string                      `json:"exe,omitempty"`
-	Gids           []int32                     `json:"gids,omitempty"`
-	IOCounters     *process.IOCountersStat     `json:"iocounters,omitempty"`
-	IsRunning      bool                        `json:"isrunning,omitempty"`
-	MemInfo        *process.MemoryInfoStat     `json:"meminfo,omitempty"`
-	MemMaps        *[]process.MemoryMapsStat   `json:"memmaps,omitempty"`
-	MemPercent     float32                     `json:"mempercent,omitempty"`
-	Name           string                      `json:"name,omitempty"`
-	NetIOCounters  []nethw.IOCountersStat      `json:"netiocounters,omitempty"`
-	Nice           int32                       `json:"nice,omitempty"`
-	NumCtxSwitches *process.NumCtxSwitchesStat `json:"numctxswitches,omitempty"`
-	NumFds         int32                       `json:"numfds,omitempty"`
-	NumThreads     int32                       `json:"numthreads,omitempty"`
-	PageFaults     *process.PageFaultsStat     `json:"pagefaults,omitempty"`
-	Parent         int32                       `json:"parent,omitempty"`
-	Ppid           int32                       `json:"ppid,omitempty"`
-	Rlimit         []process.RlimitStat        `json:"rlimit,omitempty"`
-	Status         string                      `json:"status,omitempty"`
-	Tgid           int32                       `json:"tgid,omitempty"`
-	Times          *cpu.TimesStat              `json:"cputimes,omitempty"`
-	Uids           []int32                     `json:"uids,omitempty"`
-	Username       string                      `json:"username,omitempty"`
+	Pid             int32                       `json:"pid"`
+	Background      bool                        `json:"background,omitempty"`
+	CPUPercent      float64                     `json:"cpupercent,omitempty"`
+	Children        []int32                     `json:"children,omitempty"`
+	CmdLine         string                      `json:"cmd,omitempty"`
+	ConnectionCount int                         `json:"connection_count,omitempty"`
```

Comment:
does version need to be updated here?

Reply:
yes @kannappanr this is a breaking change.

Reply:
Well hold on. We should never remove a field. It might be better to stop filling that field. Why make a breaking change when it is not required to do so? Just make that field `nil`. @anjalshireesh @harshavardhana

Reply:
@kannappanr @harshavardhana @wlan0 

It did occur to me that this could be a breaking change and my first instinct was also to set `Connections` to `empty slice`. However, I tried to confirm that and found that it should not break anything. (Hence the reference to testing with existing/older version of `mc` in the `How to test this PR?` section of the PR)

- It doesn't break compilation of `mc` as `mc` doesn't use the removed field (`Connections`) from this struct anywhere. My understanding here is that this is a purely `subnet health` related struct. `mc` gets the data from the api and then just dumps it into the health report without reading/processing the fields from the structs. So `mc` compilation is mainly dependent on the root level structs into which it deserializes the api output and not the nested structs/fields like this one.
- Doesn't break `mc admin subnet health` on existing/older versions of `mc` becasue
  - `Connections` field not referred in `mc`
  - Doesn't break the JSON deserialization of the api output because of the `omitempty`

My assumption here being that `mc` is the only client of this api, and this particular struct/field is only used for the `mc admin subnet health` command.

I have tested the following with `master` branch and three recent releases of `mc` against `minio` running with these changes:
 - execution of `mc admin subnet health` using the released version of the `mc` binary
 - compilation of `mc` with these changes
 - execution of `mc admin subnet health` using the `mc` binary compiled with these changes

Having said this, if the consensus is that it is still better to avoid removing the field, I am happy to go with any of the other options.

Apart from the two options already discussed above (set to nil or update version), another option is to make this a purely `mc` side change. `mc` transforms the api output to the latest health report format before saving it to a file. So it is possible to remove the connections information as part of that transformation as well.

Reply:
Okay if that is well tested with older mc there is no problem in removing @anjalshireesh